runtime/drivers/clickhouse/olap.go

```diff
@@ -250,7 +250,57 @@ func (c *connection) InsertTableAsSelect(ctx context.Context, name, sql string,
 			LongRunning: true,
 		})
 	}
-
+	if strategy == drivers.IncrementalStrategyReplace {
+		// create temp table with the same schema
+		tempName := tempName(name)
+		// drop the temp table
+		defer func() {
+			err := c.DropTable(ctx, tempName, false)
+			if err != nil {
+				c.logger.Error("clickhouse: failed to drop temp table", zap.String("name", tempName), zap.Error(err))
+			}
+		}()
+		err := c.Exec(ctx, &drivers.Statement{
+			Query:    fmt.Sprintf("CREATE TABLE %s AS %s", safeSQLName(tempName), name),
+			Priority: 1,
+		})
+		if err != nil {
+			return err
+		}
+		// insert into temp table
+		err = c.Exec(ctx, &drivers.Statement{
+			Query:    fmt.Sprintf("INSERT INTO %s %s", safeSQLName(tempName), sql),
+			Priority: 1,
+		})
+		if err != nil {
+			return err
+		}
+		// list partitions from the temp table
+		res, err := c.Execute(ctx, &drivers.Statement{
```

Comment:
I think it should use `partition` columns from `system.parts` table to list partition table. 
https://clickhouse.com/docs/en/sql-reference/statements/alter/partition#how-to-set-partition-expression

Reply:
Agree.

Reply:
is `system.parts` strong or eventually consistent?

Reply:
I think it does not matter. The data is being considered as ingested into temp table only after the partition key is added to the parts table. [See here](https://arc.net/l/quote/rhgwrgoe)